src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisResultService.java

```diff
@@ -90,21 +98,63 @@ private Sort createNResultSort(Sort sort) {
     }
 
     @Transactional(readOnly = true)
-    public Page<ContingencyResultDTO> findNmKContingenciesResult(UUID resultUuid, String stringFilters, Pageable pageable) {
+    public Page<ContingencyResultDTO> findNmKContingenciesPaged(UUID resultUuid, String stringFilters, Pageable pageable) {
         assertResultExists(resultUuid);
 
         Page<ContingencyEntity> contingencyPageBis = self.findContingenciesPage(resultUuid, fromStringFiltersToDTO(stringFilters), pageable);
         return contingencyPageBis.map(ContingencyResultDTO::toDto);
     }
 
     @Transactional(readOnly = true)
-    public Page<SubjectLimitViolationResultDTO> findNmKConstraintsResult(UUID resultUuid, String stringFilters, Pageable pageable) {
+    public List<ContingencyResultDTO> findNmKContingenciesResult(UUID resultUuid) {
+        assertResultExists(resultUuid);
+
+        List<ContingencyEntity> contingencies = contingencyRepository.findAllByResultId(resultUuid);
+        List<UUID> uuids = contingencies.stream().map(contingency -> contingency.getUuid()).toList();
+        contingencyRepository.findAllWithContingencyElementsByUuidIn(uuids);
+        contingencyRepository.findAllWithContingencyContingencyLimitViolationsByUuidIn(uuids);
```

Comment:
Is this to prevent n+1 query ? Could you add a comment if this is the case

Reply:
done, renamed findAllWithContingencyContingencyLimitViolationsByUuidIn method as well